### ISSUE TITLE
Compile with Syntax Error Throws NRE not ScriptEngineException

### ISSUE THREAD

**Issue by escape-llc** [Microsoft/ClearScript#66]
I recently tried updating to 5.5.2.0 and there were some problems compiling expressions containing syntax errors, all my test cases suddenly failed in an unexpected way!  I am using V8 as the engine.

The call to `Runtime.Compile` throws `NullReferenceException` instead of `ScriptEngineException`.  Not sure if there's some binary interop problems with the V8 assemblies, but all other scripting function work as expected.

Syntactically correct expressions have no problems and evaluate correctly.

When i regressed to 5.5.1.0, everything went back to working.

Below is some Immediate Window output for reference.  I simply set a BP on my `Runtime.Compile` and executed it in the IW instead:

```
5.5.2.0
Runtime
{Microsoft.ClearScript.V8.V8Runtime}
    FormatCode: false
    HeapSizeSampleInterval: {00:00:00}
    MaxHeapSize: {0}
    MaxStackUsage: {0}
    Name: "V8Runtime"
js
"doc.Property1 == Value1'"
Runtime.Compile(js)
'Runtime.Compile(js)' threw an exception of type 'System.NullReferenceException'
    Data: {System.Collections.ListDictionaryInternal}
    HResult: -2147467261
    HelpLink: null
    InnerException: null
    Message: "Object reference not set to an instance of an object."
    Source: "ClearScriptV8-64"
    StackTrace: "   at V8Exception.ThrowScriptEngineException(V8Exception* )\r\n   at Microsoft.ClearScript.V8.V8IsolateProxyImpl.Compile(String gcDocumentName, String gcCode)"
    TargetSite: {Void V8Exception.ThrowScriptEngineException(V8Exception*)}

Just to show "normal" expressions compile just fine:
Runtime.Compile("doc.Property1=='value1'")
{Microsoft.ClearScript.V8.V8ScriptImpl}
    Name: "V8Runtime:Script Document [2]"

5.5.1.0
Runtime
{Microsoft.ClearScript.V8.V8Runtime}
    FormatCode: false
    HeapSizeSampleInterval: {00:00:00}
    MaxHeapSize: {0}
    MaxStackUsage: {0}
    Name: "V8Runtime"
js
"doc.Property1 == Value1'"
Runtime.Compile(js)
'Runtime.Compile(js)' threw an exception of type 'Microsoft.ClearScript.ScriptEngineException'
    Data: {System.Collections.ListDictionaryInternal}
    EngineName: ""
    ErrorDetails: "SyntaxError: Invalid or unexpected token\n    at V8Runtime:Script Document:1:24 -> doc.Property1 == Value1'"
    ExecutionStarted: false
    HResult: -2146233079
    HelpLink: null
    InnerException: null
    IsFatal: false
    Message: "SyntaxError: Invalid or unexpected token"
    Source: "ClearScriptV8-64"
    StackTrace: "   at V8Exception.ThrowScriptEngineException(V8Exception* )\r\n   at Microsoft.ClearScript.V8.V8IsolateProxyImpl.Compile(String gcDocumentName, String gcCode)"
    TargetSite: {Void V8Exception.ThrowScriptEngineException(V8Exception*)}

```

Interestingly, both versions seem to be at the same place in the call stack, but something is now missing.

**Comment by ClearScriptLib** [comment]
Hi @escape-llc,

Could you post a minimal JavaScript code sample that triggers the exception?

Also, each newer ClearScript release uses a significantly updated V8 release under the covers. Could you try Version 5.5.3? There's now an [official NuGet package](https://www.nuget.org/packages/Microsoft.ClearScript).

Thanks!

**Comment by c-s-n** [comment]
Hi @ClearScriptLib,

I can confirm that this issue is still present in version 5.5.3, obtained via NuGet.

`V8Runtime` and `V8ScriptEngine` have different behavior currently:
```C#
using (var runtime = new V8Runtime(V8RuntimeFlags.None))
{
	runtime.Compile(@"function )}"); // throws NullReferenceException -> wrong
}
using (var engine = new V8ScriptEngine(V8ScriptEngineFlags.None))
{
	engine.Compile(@"function )}"); // throws ScriptEngineException -> correct
}
```

**Comment by ClearScriptLib** [comment]
Confirmed. Thank you!

**Comment by ClearScriptLib** [comment]
Fix posted [here](https://github.com/Microsoft/ClearScript/commit/ea75e470bb1f3524bd407e9da7125429fcb994c6).

**Comment by ClearScriptLib** [comment]
Fixed in [Version 5.5.4](https://github.com/Microsoft/ClearScript/releases/tag/5.5.4).
